pkg/scm/keys.go

```diff
@@ -0,0 +1,104 @@
+package scm
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"golang.org/x/crypto/ssh"
+	"github.com/pluralsh/plural/pkg/utils"
+	homedir "github.com/mitchellh/go-homedir"
+	"path/filepath"
+)
+
+func generateKeys() (pub string, priv string, err error) {
+	pub, priv, found := readKeys()
+	if found {
+		return
+	}
+
+	pubKey, privKey, err := ed25519.GenerateKey(rand.Reader)
+	if err != nil {
+		return
+	}
+
+	b, err := x509.MarshalPKCS8PrivateKey(privKey)
+	if err != nil {
+		return
+	}
+
+	priv = string(pem.EncodeToMemory(&pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: b,
+	}))
+
+	sshPub, err := ssh.NewPublicKey(pubKey)
+	if err != nil {
+		return
+	}
+	pub = string(ssh.MarshalAuthorizedKey(sshPub))
+
+	err = saveKeys(pub, priv)
+	return
+}
+
+func readKeys() (pub string, priv string, found bool) {
+	pubFile, privFile, err := keyFiles()
+	if err != nil {
+		return
+	}
+
+	if !utils.Exists(pubFile) || !utils.Exists(privFile) {
+		return
+	}
+
+	pub, err = utils.ReadFile(pubFile)
+	if err != nil {
+		return
+	}
+
+	priv, err = utils.ReadFile(privFile)
+	if err != nil {
+		return
+	}
+
+	found = true
+	return
+}
+
+func saveKeys(pub, priv string) error {
+	if !utils.Confirm("Would you like to save the keys to ~/.ssh?") {
+		return nil
+	}
+
+	pub, priv, err := keyFiles()
```

Comment:
I might be misreading this but it looks like you're getting the filepath strings from `keyFiles()` here and saving the path as the byte array in the file instead of the generated key. Which would explain why the files generated for me were just path names.

![Screen Shot 2022-05-17 at 11 16 40 PM](https://user-images.githubusercontent.com/33042053/168970228-4cdcc227-face-442c-9aeb-418f87377171.png)

It seems like the contents of the keys are just their own directory paths?

![Screen Shot 2022-05-17 at 11 31 06 PM](https://user-images.githubusercontent.com/33042053/168972548-536cdf14-e11b-4204-a14d-0f63672cc98c.png)

Reply:
Oh shit this might all be buggy, I didn't test it thoroughly